### PR TITLE
Spatial_searching doc : Update Search_traits_3 requirements

### LINKED ISSUE #261: Update search Search_traits_3 requirements
The `AABB_tree<AT>` internally use the spatial searching and in particular instantiates Search_traits_3 with AT. This implies according to that page that the concept `AABBTraits` must be a Kernel.

A clean solution is adding a new concept in Spatial searching that describes requirements of the template parameter of Search_traits_3. Then we can say that AABBTraits refines this concept. For consistency, we do the same for the template parameter of Search_traits_2.

### PR DESCRIPTION
**Pull request by janetournois**
fixes #261 

this PR implements what is listed in the issue

@sloriot can you please check whether this PR is complete?

### CODE REVIEW

File: Spatial_searching/doc/Spatial_searching/Concepts/SearchGeomTraits_2.h [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/*!
+\ingroup PkgSpatialSearchingDConcepts
+\cgalConcept
+
+The concept `SearchGeomTraits_2` defines the requirements for the template 
+parameter of the search traits classes.
+
+\cgalHasModel `Simple_cartesian<double>`
+\cgalHasModel `Simple_cartesian<Gmpq>`
```
Comment (sloriot) [inline Spatial_searching/doc/Spatial_searching/Concepts/SearchGeomTraits_2.h]: You can simply say "Any \cgal Kernel"

**Comment by sloriot** [comment]
Looks good!

**Comment by janetournois** [comment]
Thanks. It will be tested in [CGAL-4.9-Ic-134](https://cgal.geometryfactory.com/CGAL/testsuite/results-4.9-Ic-134.shtml)

**Comment by lrineau** [comment]
In CGAL-4.9-Ic-134, the Doxygen testsuite results were good.

**Comment by lrineau** [comment]
This PR has been merged in `master` (targeting CGAL-4.9).

**Pull request merged**